test/jdk/java/net/httpclient/lib/jdk/httpclient/test/lib/http2/Http2TestServerConnection.java

```diff
@@ -960,13 +960,29 @@ private void handlePush(OutgoingPushPromise op) throws IOException {
         final BodyOutputStream oo = new BodyOutputStream(
                 promisedStreamid,
                 clientSettings.getParameter(
-                        SettingsFrame.INITIAL_WINDOW_SIZE), this);
+                        SettingsFrame.INITIAL_WINDOW_SIZE), this) {
+
+            @Override
+            protected void sendEndStream() {
```

Comment:
The method signature of `BodyOutputStream`'s `sendEndStream()` method allows for `IOException` to be thrown. So I think we should just do the same here (i.e. `protected void sendEndStream() throws IOException`). That way you can remove the try/catch block in this method which currently just prints the exception stacktrace. The caller can then decide what to do with that exception, which in the current case, the `BodyOutputStream` just logs.

Reply:
Agreed, well spotted. Much nicer to have it as `protected void sendEndStream() throws IOException`.